src/components/SlateEditor/plugins/related/EditRelated.jsx

```diff
@@ -100,13 +100,10 @@ class EditRelated extends React.PureComponent {
   }
 
   async searchForArticles(inp) {
-    const articles = (
-      await searchDrafts({
-        query: inp,
-        fallback: true,
-      })
-    ).results;
-    return articles ? articles.filter(article => !!article.id) : [];
+    return await searchDrafts({
```

Comment:
Koden burde fungere fra før av også, for den gjør sjekken i AsyncDropdown om det er en array og ekstrakter .results hvis den får en SearchObject `(Array.isArray(apiOutput) ? apiOutput : apiOutput.results) || [];` så da kunne ikke første delen av koden være feil. Er det da filter som fant artikler uten id og returnerte tom array? isåfall så er det vel bra, for vi vil ikke håndtere "ugyldige" artikler.

Reply:
Problemet var ikkje ugyldige artikler men at søk etter relaterte artikler alltid returnerte at det var 10 treff. Filter på at artikler har id gir ikkje meining fordi det ikkje finnes lagra artikler som ikkje har id.

Reply:
Jeg ser nå en feil i AsyncDropdown med at den tar ikke utgangspunkt i lengden dersom den får en array istedenfor objekt, den defaulter til `null` hvis den ikke finner objekt.totalCount, så det burde også fikses siden det er ikke alltid vi sender den searchObjekter men arrayer. Så i teorien kunne bare den lille endringen fikset det problemet du hadde, men siden vi kan aldri få artikler uten id så er det vel like greit å fjerne redundante filteret. Legger du på en ekstra `|| apiOutput.length` på linje 65 i AsyncDropdown.jsx i samme slenge, eller skal den gjøres i separat pr?

Reply:
Nvm, my bad, siden vi gjør paginering søket, henter vi uansett ikke mer enn 10 resultater per request, så å gjøre apiOutput.length er redundant siden `null` blir gjort om til `items.length` ned i kjeden.